### DISCUSSION TITLE
go sdk v1.27 初始化的 组件无法使用

### DISCUSSION THREAD

**Discussion by zxl199409** [aws/aws-sdk-go-v2#2669] [Q&A] [answered]
之前是采用的 github.com/aws/aws-sdk-go-v2 v1.17.3 // indirect

var Client *dynamodb.Client

func init() {
	region := cfg.EnvVar.Region
	awsConfig, err := config.LoadDefaultConfig(context.TODO(),
		config.WithRegion(region))

	if err != nil {
		panic(err)
	}

	awsv2.AWSV2Instrumentor(&awsConfig.APIOptions)

	endpoint := cfg.EnvVar.DynamodbEndpoint
	if endpoint != "" {
		endpointResolver := dynamodb.WithEndpointResolver(dynamodb.EndpointResolverFromURL(endpoint))
		Client = dynamodb.NewFromConfig(awsConfig, endpointResolver)
	} else {
		Client = dynamodb.NewFromConfig(awsConfig)
	}

}

升级版本后无法使用

**Comment by zxl199409** [comment]
error not found, ResolveEndpointV2

**Comment by lucix-aws** [comment, marked as answer]
#2370
